icu4c/source/common/ubiditransform.cpp

```diff
@@ -499,7 +499,8 @@ ubiditransform_transform(UBiDiTransform *pBiDiTransform,
     /* Checking for U_SUCCESS() within the loop to bail out on first failure. */
     for (action = pBiDiTransform->pActiveScheme->actions; *action && U_SUCCESS(*pErrorCode); action++) {
         if ((*action)(pBiDiTransform, pErrorCode)) {
-            if (action + 1) {
+            const UBiDiAction * next = action+1;
```

Comment:
It looks like the original code may be incorrect; `(action + 1)` can never be null here. Probably what was intended was `if (*(action + 1)) {`, but this would change the behavior, so check with someone who knows this code.